packages/cli/cli/src/cli.ts

```diff
@@ -361,19 +360,40 @@ function addDiffCommand(cli: Argv<GlobalCliOptions>, cliContext: CliContext) {
                     string: true,
                     description: "The previous version of the API (e.g. 1.1.0)"
                 })
+                .option("from-generator-version", {
+                    string: true,
+                    description: "The previous version of the generator (e.g. 1.1.0)"
+                })
+                .option("to-generator-version", {
+                    string: true,
+                    description: "The next version of the generator (e.g. 1.1.0)"
+                })
                 .option("quiet", {
                     boolean: true,
                     default: false,
                     alias: "q",
                     description: "Whether to suppress output written to stderr"
+                })
+                .middleware((argv) => {
+                    if (!haveSameNullishness(argv.fromGeneratorVersion, argv.toGeneratorVersion)) {
+                        throw new Error(
+                            "Both --from-generator-version and --to-generator-version must be provided together, or neither should be provided"
+                        );
+                    }
                 }),
         async (argv) => {
-            const fromVersion = argv.fromVersion != null ? argv.fromVersion : undefined;
+            const fromVersion = undefinedIfNullish(argv.fromVersion);
```

Comment:
call me a boomer but I'm more of a `argv.fromVersion != null ? argv.fromVersion : undefined`

But maybe it's nice to use the function since you use the other function right below